### DISCUSSION TITLE
what is the difference between ipxe and edk2

### DISCUSSION THREAD

**Discussion by ludezhen** [ipxe/ipxe#1144] [Q&A]
hi, ipxe guys:
i am very confused about the difference between ipxe and edk2. when we talk about uefi, it is seems like that uefi is equals to edk2, so when i find ipxe is also support uefi，i am not sure whether ipxe code can replace edk2 code  and it can also works well in uefi mode.

**Comment by mcb30** [comment]
EDK2 is the reference implementation of the UEFI firmware.  iPXE is a network bootloader that can run inside the UEFI environment.

iPXE can therefore replace the portions of EDK2 that deal with network boot.  It can also interoperate with these code portions: for example, you can use the EDK2 network stack to download and run iPXE, which then uses its own network stack to download and boot your operating system.

**Reply by NiKiZe** [reply]
To add to this, iPXE has several features that is out of scope of basic firmware to deal with.
